pkg/registry/populator.go

```diff
@@ -302,15 +302,11 @@ func (i *DirectoryPopulator) getNextReplacesImagesToAdd(imagesToAdd []*ImageInpu
 		// no new images can be added, the current iteration aggregates all the
 		// errors that describe invalid bundles
 		if pkgFoundImages == 0 && pkgRemainingImages > 0 {
```

Comment:
nit: you could always use the lengths of `remainingImages` and `foundImages` instead of these counters

Reply:
+1 just didn't want to change too much about the add when implementing the overwrite fix